### ISSUE TITLE
False positives for both SC2034 and SC2102 with range expressions

### ISSUE THREAD

**Issue by automorphism88** [koalaman/shellcheck#2027]
#### For bugs
- Rule Id (if any, e.g. SC1000): SC2034, SC2102
- My shellcheck version (`shellcheck --version` or "online"): 0.7.1 and online
- [x] The rule's wiki page does not already cover this (e.g. https://shellcheck.net/wiki/SC2086)
The wiki page for SC2102 mentions this, but not the page for SC2034
- [x] I tried on https://www.shellcheck.net/ and verified that this is still a problem on the latest commit

#### Here's a snippet or screenshot that shows the problem:

```sh

#!/bin/bash
set -eu
shopt -s extglob
readonly AAA=abc
case ${1-} in
    x*([$AAA]))
        echo match ;;
    *)
        echo no match ;;
esac

```

#### Here's what shellcheck currently says:

```
Line 4:
readonly AAA=abc
         ^-- SC2034: AAA appears unused. Verify use (or export if used externally).
 
Line 6:
    x*([$AAA]))
       ^-- SC2102: Ranges can only match single chars (mentioned due to duplicates).
```

#### Here's what I wanted or expected to see:

No output, the script is valid and works as intended. Note that enclosing the variable expansion in double quotes will avoid this issue, but in this case, with a readonly variable that contains no whitespace, there is no need for double quotes.

**Comment by drjasonharrison** [comment]
Line 6: The problem isn't the variable, it's the unquoted `[...]`

The expression `$AAA` is interpreted as a range, not a variable reference.

Maybe this will work
 
```
x*("\[$AAA\]"))
```
